jenkinsfile-examples/cross-platform-build/Jenkinsfile

```diff
@@ -0,0 +1,174 @@
+            stash name: 'YOUR_CROSSPLATFORM_SOURCES'
+        }
+    }
+
+    //
+    // #2
+    // Build on each platform.
+    //
+    stage('Build Cross-Platform Libraries') {
+        timeout(5) {
+            parallel (
+                "win32-stream" : {
+                    node('win32') {
+                        // This will copy all files packaged in YOUR_CROSSPLATFORM_SOURCES to agent workspace root directory.
+                        // To copy to another agent directory, see [https://github.com/BillHoo/pipeline-examples/tree/master/pipeline-examples/unstash-different-dir]
+                        unstash 'YOUR_CROSSPLATFORM_SOURCES'
+                        
+                        // Change current directory to "build", and run CMake.
+                        // NOTE we assume your CMake INSTALL target will put output binaries into 'dist/win32' for later usage.
+                        dir('build') {
+                            bat 'cmake ..\\src'
+                            bat 'cmake --build . --config RelWithDebInfo --target INSTALL'   
+                        }
+                    }
+                },
+            
+                "android-stream" : {
+                    node('android') {
+                        unstash 'YOUR_CROSSPLATFORM_SOURCES'
+                        
+                        // Change current directory to "build", and run Android build script.
+                        dir('build') {
+                            sh './android-build.sh'
+                        }
+                    }
+                },
+            
+                "ios-stream" : {
+                    node('ios') {
+                        unstash 'YOUR_CROSSPLATFORM_SOURCES'
+                        
+                        // Change current directory to "build", and run iOS build script.
+                        dir('build') {
+                            sh './iphoneos-build.sh'
+                        }
+                    }
+                },
+            
+                "mac-stream" : {
+                    node('mac') {
+                        unstash 'YOUR_CROSSPLATFORM_SOURCES'
+                        
+                        // Change current directory to "build", and run MacOSX build script.
+                        dir('build') {
+                            sh './macos-build.sh'
+                        }
+                    }
+                }
+            ) // parallel
+        } // Timeout
+    } // State cross-platform build
+    
+    //
+    // #3
+    // Cross-Platform Libraries Distribution
+    //
+    stage('Libraries Distribution') {
+        parallel(
+            "win32-archiving" : {
+                node('win32') {
+                    step([$class: 'ArtifactArchiver', artifacts: 'dist\\win32\\*', fingerprint: true])
+                    step([$class: 'ArtifactArchiver', artifacts: 'symbols\\win32\\*', fingerprint: true])
```

Comment:
This verbose syntax is no longer required; `archiveArtifacts artifacts: '**/foo.bar', fingerprint: true` should be sufficient.